omas/omas_physics.py

```diff
@@ -261,11 +261,9 @@ def summary_lineaverage_density(ods, line_grid=2000, time_index=None, update=Tru
             i1 = zero_crossings[0]
             i2 = zero_crossings[-1]
 
-            ne_line = 0.
-            for i in range(i1, i2, numpy.sign(i2-i1)):
-                psival = psi_interp(Zline[i], Rline[i])[0]
-                ne_interp  = scipy.interpolate.splev(psival, tck)
-                ne_line  += ne_interp
+            psival = [psi_interp(Zline[i], Rline[i])[0] for i in range(i1, i2, numpy.sign(i2-i1))]
+            ne_interp  = scipy.interpolate.splev(psival, tck)
+            ne_line  = numpy.trapz(ne_interp)
```

Comment:
@jmcclena does use of trapz allow fewer points to be used along the interf line of sight?

Reply:
@eldond I'm not quite sure what you mean.

Reply:
You were interpolating to get a bunch of points sampling the line and then adding all the values together. Presumably using trapezoid integration gives higher accuracy than boxes for the same number of points, or the same accuracy for fewer points (making it faster). Also, does this work as expected?

Reply:
Oh, I see. Yes, I'm getting very similar results. Presumably trapz is more accurate. I'm getting slightly faster for the new algorithm.